src/electron/windows/jsign.mjs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {spawn} from 'child_process';
+import {resolve} from 'path';
+
+/**
+ * Get the full path of jsign.jar.
+ * @returns {string} The full path similar to <.../jsign.jar>.
+ */
+function getJsignJarPath() {
+  return resolve('third_party', 'jsign', 'jsign-4.0.jar');
+}
+
+/**
+ * Run jsign.jar according to the corresponding options targeting fileToSign.
+ * @param {string} fileToSign The path string of a file to be signed.
+ * @param {string[]} options The options to be passed to jsign. see https://ebourg.github.io/jsign/
+ * @returns {Promise<number>} A promise containing the exit code of jsign.
+ */
+export default async function jsign(fileToSign, options) {
+  if (!options) {
+    throw new Error('options are required by jsign');
+  }
+  if (!fileToSign) {
+    throw new Error('fileToSign is required by jsign');
+  }
+
+  const jsignProc = spawn(
+      'java',
```

Comment:
Humm, can you actually test if java is available. It would be nice to alert the user that java is needed, since they may not have that setup. I don't know what error you'll get otherwise.

Reply:
It will print out the following error in CLI:

```
Error: spawn java ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn java',
  path: 'java',
  ...
```

Reply:
Not a great error. Better to check is java exists, though I guess it's not easy to check the PATH...
Is errno = -2 always the executable missing? In that case we could check errno and output a better message

Reply:
Agree, and the error will leak command line arguments passed to `jsign` (which might contain our GCP credentials in the future). It'd better to try-catch the error instead of leveraging the default behavior.

Reply:
added `jsignProc.on('error' ...)` to handle errors caused by "not able to launch `java`".